pkg/broker/handler/processors/deliver/processor_test.go

```diff
@@ -321,6 +311,94 @@ func TestDeliverFailure(t *testing.T) {
 	}
 }
 
+type NoReplyHandler struct{}
+
+func (NoReplyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	w.WriteHeader(http.StatusAccepted)
+	// Read body to allow the connection to be reused.
```

Comment:
May I get a reference link? Just curious how it works.

Reply:
Sure, this is documented in the [HTTP Response](https://golang.org/pkg/net/http/#Response) Body field:

> The http Client and Transport guarantee that Body is always
> non-nil, even on responses without a body or responses with
> a zero-length body. It is the caller's responsibility to
> close Body. The default HTTP client's Transport may not
> reuse HTTP/1.x "keep-alive" TCP connections if the Body is
 > not read to completion and closed.

I am currently working on converting HTTP client code to use CE bindings which will give us direct control over this behavior in the deliver code as well as I'm not sure that pubsub transport always handles this properly.

Reply:
On second look this only applies to response bodies, not request bodies, so this is not actually necessary (but shouldn't hurt).